src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { Loader } from '../Loader/Loader';
+
+type Props = {
+  todo: Todo,
+  deleteTodo: (todoId: number) => Promise<void>,
+  updateTodoStatus: (todo: Todo, status: boolean) => Promise<void>,
+  updateTodoTitle: (todo: Todo, title: string) => Promise<void>,
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  deleteTodo,
+  updateTodoStatus,
+  updateTodoTitle,
+}) => {
+  const [inputText, setInputText] = useState(todo.title);
+  const [isChangeTitle, setIsChangeTitle] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const titleRef = useRef<HTMLInputElement>(null);
+
+  const toDefaultTitle = () => {
+    setInputText(todo.title);
+    setIsChangeTitle(false);
+  };
+
+  useEffect(() => {
+    if (titleRef.current && isChangeTitle) {
+      titleRef.current.focus();
+    }
+  }, [titleRef.current, isChangeTitle]);
+
+  const handleKeyDown = (event: { key: string; }) => {
```

Comment:
Are you sure that it's the right type for this event? It's most likely has its own type. Try to find the correct type for onKeyDown event